src/providers/mod.rs

```diff
@@ -130,7 +132,18 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
                 format!("failed to write to file {:?}", temp_file.path().display())
             })?;
         }
-
+        let name = user.name().to_string_lossy();
+        let path = file_path.to_string_lossy();
+        let mut map: HashMap<&str, &str> = HashMap::new();
```

Comment:
See https://docs.rs/maplit/1.0.2/maplit/macro.hashmap.html for a cleaner way of doing this.